### PR TITLE
“Loadbalancer”

### PR DESCRIPTION
**Pull request by amitnayak1505**

### CODE REVIEW

File: alb.tf [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    ipv6_cidr_blocks = ["::/0"]
+  }
+
+  egress {
+    from_port        = 0
+    to_port          = 0
+    protocol         = "-1"
+    cidr_blocks      = ["0.0.0.0/0"]
+    ipv6_cidr_blocks = ["::/0"]
+  }
```
Comment (vladimirmukhin) [inline alb.tf]: You don't need 22 and 443 for your loadbalancer. You can keep only 80

File: remotestate.tf [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  name = "${var.aws_dynamodb_table}"
+  read_capacity = 20
+  write_capacity = 20
+  hash_key = "LockID"
+
+  attribute {
+      name = "LockID"
+      type = "S"
+  }
+}
```
Comment (vladimirmukhin) [inline remotestate.tf]: Please remove all files that do not relate to the load balancer task

File: target_group.tf [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    interval            = 30
+    path                = "/"
+    protocol            = "HTTP"
+    matcher             = 200
+}
+
+lifecycle {
+    create_before_destroy = true
+  }
+}
```
Comment (vladimirmukhin) [inline target_group.tf]: There is no need to have it as a separate file. Please merge it with alb.tf

**Review by vladimirmukhin — CHANGES_REQUESTED** [review]

**Review by vladimirmukhin — CHANGES_REQUESTED** [review]
I see a lot of code style issues in this PR. Please take time to fix all of them.
https://www.terraform.io/language/syntax/style
https://www.terraform-best-practices.com/

File: remotestate.tf (hunk 2 above)
Comment (amitnayak1505) [reply remotestate.tf]: Done

File: target_group.tf (hunk 3 above)
Comment (amitnayak1505) [reply target_group.tf]: merged with alb.tf

File: alb.tf [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+    Name = "sglb"
+  }
+
+  lifecycle {
+    create_before_destroy = true
+  }
+}
+
+
+resource "aws_lb_target_group" "sample_tg" {
```
Comment (vladimirmukhin) [inline alb.tf]:
```suggestion
}

resource "aws_lb_target_group" "sample_tg" {
```

File: alb.tf [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+
+  health_check {
+    healthy_threshold   = 2
+    unhealthy_threshold = 2
+    interval            = 30
+    path                = "/"
+    protocol            = "HTTP"
+    matcher             = 200
+
+  }
```
Comment (vladimirmukhin) [inline alb.tf]:
```suggestion
    matcher             = 200
  }
```

File: alb.tf [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+    create_before_destroy = true
+  }
+}
+
+
+resource "aws_lb" "appln-lb" {
+
+  count = length(var.public_cidr)
+
+  name               = "appln-lb"
```
Comment (vladimirmukhin) [inline alb.tf]:
```suggestion
}

resource "aws_lb" "appln-lb" {
  count = length(var.public_cidr)

  name               = "appln-lb"
```

File: alb.tf [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+
+  enable_deletion_protection = false
+
+  tags = {
+    Environment = "${var.env_code}-appln-lb"
+  }
+}
+
+
+resource "aws_lb_listener" "listner" {
```
Comment (vladimirmukhin) [inline alb.tf]:
```suggestion
}

resource "aws_lb_listener" "listner" {
```

File: alb.tf [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+      status_code  = "200"
+    }
+  }
+
+  depends_on = [aws_lb.appln-lb]
+}
+
+resource "aws_lb_listener_rule" "rule" {
+
+  count = 2
```
Comment (vladimirmukhin) [inline alb.tf]:
```suggestion
resource "aws_lb_listener_rule" "rule" {
  count = 2
```

Comment (vladimirmukhin) [inline alb.tf]: You don't need conunt here

File: alb.tf [hunk 9]
```diff
@@ -0,0 +1,118 @@
...
+
+  lifecycle {
+    create_before_destroy = true
+  }
+}
+
+
+resource "aws_lb" "appln-lb" {
+
+  count = length(var.public_cidr)
```
Comment (vladimirmukhin) [inline alb.tf]: You don't need count here

**Review by vladimirmukhin — CHANGES_REQUESTED** [review]

File: alb.tf (hunk 9 above)
Comment (amitnayak1505) [reply alb.tf]: removing the count

File: alb.tf (hunk 8 above)
Comment (amitnayak1505) [reply alb.tf]: removed the count

**Review by vladimirmukhin — APPROVED** [review]

**Pull request merged**